tests/UrlGeneratorTest.php

```diff
@@ -671,6 +733,22 @@ public function testGenerateFromCurrent(): void
         $this->assertEquals('/ru/home/index', $url);
     }
 
+    public function testGenerateFromCurrentWithQueryString(): void
+    {
+        $request = new ServerRequest('GET', 'http://example.com/en/home/index?test=1');
+        $route = Route::get('/{_locale}/home/index')->name('index');
+
+        $currentRoute = new CurrentRoute();
+        $currentRoute->setUri($request->getUri());
+        $currentRoute->setRouteWithArguments($route, ['_locale' => 'en']);
+        $urlGenerator = $this->createUrlGenerator([$route], $currentRoute);
+        $urlGenerator->setDefaultArgument('_locale', 'uz');
+
+        $url = $urlGenerator->generateFromCurrent(['_locale' => 'ru']);
+
+        $this->assertEquals('/ru/home/index?test=1', $url);
```

Comment:
Don't agree that it's correct and expected behaviour

Reply:
You suggest reset any GET-params? Why?

Expected behaviour for me: get equal url with replaced locale only.

Reply:
Agree with that test. Missed method name when read the code.